internal/services/oracle/oracle_network_anchor_resource.go

```diff
@@ -0,0 +1,402 @@
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.NetworkAnchors
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model NetworkAnchorResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+			id := networkanchors.NewNetworkAnchorID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			properties := &networkanchors.NetworkAnchorProperties{
+				ResourceAnchorId:                     model.ResourceAnchorId,
+				SubnetId:                             model.SubnetId,
+				IsOracleDnsForwardingEndpointEnabled: pointer.To(model.OracleDnsForwardingEndpointEnabled),
+				IsOracleDnsListeningEndpointEnabled:  pointer.To(model.OracleDnsListeningEndpointEnabled),
+				IsOracleToAzureDnsZoneSyncEnabled:    pointer.To(model.OracleToAzureDnsZoneSyncEnabled),
+			}
+
+			if model.OciVcnDnsLabel != "" {
+				properties.OciVcnDnsLabel = pointer.To(model.OciVcnDnsLabel)
+			}
+
+			if model.OciBackupCidrBlock != "" {
+				properties.OciBackupCidrBlock = pointer.To(model.OciBackupCidrBlock)
+			}
+
+			if len(model.DnsForwardingRule) > 0 {
+				properties.DnsForwardingRules = expandDnsForwardingRules(model.DnsForwardingRule)
+			}
+
+			if model.DnsListeningEndpointAllowedCidrs != "" {
+				properties.DnsListeningEndpointAllowedCidrs = pointer.To(model.DnsListeningEndpointAllowedCidrs)
+			}
+
+			param := networkanchors.NetworkAnchor{
+				Name:       pointer.To(model.Name),
+				Location:   location.Normalize(model.Location),
+				Tags:       pointer.To(model.Tags),
+				Zones:      pointer.To(model.Zones),
+				Properties: properties,
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			// Clear out create-only fields from state to avoid issue with a plan
+			if err := metadata.ResourceData.Set("dns_forwarding_rule", nil); err != nil {
+				return fmt.Errorf("cleanup `dns_forwarding_rule`: %+v", err)
+			}
+			return nil
+		},
+	}
+}
+
+func (r NetworkAnchorResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.NetworkAnchors
+			id, err := networkanchors.ParseNetworkAnchorID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model NetworkAnchorResourceModel
+			if err = metadata.Decode(&model); err != nil {
+				return err
```

Comment:
This should also follow the standard decoding error return pattern
```suggestion
				return fmt.Errorf("decoding: %+v", err)
```